net.sf.eclipsecs.core/src/net/sf/eclipsecs/core/config/meta/MetadataFactory.java

```diff
@@ -155,6 +172,136 @@ public static RuleMetadata getRuleMetadata(String name) {
     return metadata;
   }
 
+  /**
+   * Create metadata for modules not present in the previously eclipse provided metadata.
+   * Work in progress.
+   *
+   * @param moduleDetails module details fetched from checkstyle metadata
+   * @return ruleMetadata for the module
+   */
+  public static RuleMetadata createRuleMetadata(ModuleDetails moduleDetails) {
+    final String[] packageTokens = moduleDetails.getFullQualifiedName().split("\\.");
+    RuleGroupMetadata group;
+    if ((packageTokens[2]).equals("sevntu")) {
+      group = getRuleGroupMetadata("SevNTU checks");
```

Comment:
there should be no hard code of any thirdparty

Reply:
Then how would we assign this `RuleGroup` to new sevntu checks?

Reply:
it is from meta files of plugin
https://github.com/sevntu-checkstyle/sevntu.checkstyle/blob/45d83aed4b18f9f3ef93f2c9f1987d36e754debe/eclipsecs-sevntu-plugin/src/com/github/sevntu/checkstyle/checks/coding/checkstyle-metadata.xml#L1-L8

```
<checkstyle-metadata>
  <rule-group-metadata name="SevNTU checks" priority="1600">
```

extension still have to provide this information.
it is plugin specific requirements and it might be good for new `yml` config file in extension to let eclipse-cs read it and consider this.

Reply:
Okay, why not let eclipse cs store an yml file containg list of 3rd party checks, and their rule name?

Reply:
eclipse-cs is not aware of all thirdparties.
There are numerous extensions like "sevntu" we do not know them. It use to work like this for many many years ans will continue to execute like this. Eclipse-cs take all details on Checks from another jar.

Reply:
Okay then I'll send a sevntu PR to have a yml config, and eclipse-cs should have an yml parsing util

Reply:
yes.
or you can have them as xml, if you want.

Reply:
@romani Done

Reply:
merged.